### PR TITLE
179 gitaly report finalizer

### PR DESCRIPTION
**Pull request by ivanbukhtiyarov**
For cqfn#210 and cqfn#179
Complex document refactoring: added intro, virtual storage info, structure refactoring and so on

### CODE REVIEW

File: reports/gitlab-commit-protocol/gitlab-commit-protocol.tex [hunk 1]
```diff
@@ -65,6 +69,18 @@
...
+
+Gitlab team pays attention on fact that number of users, repositories, and activity grows, it is important to scale Gitaly appropriately by:
+
+\begin{enumerate}
+\item Increasing the available CPU and memory resources available to Git before resource exhaustion degrades Git, Gitaly, and GitLab application performance.
+\item Increase available storage before storage limits are reached causing write operations to fail.
+\item Improve fault tolerance by removing single points of failure. Git should be considered mission critical if a service degradation would prevent you from deploying changes to production.
+\end{enumerate}
+
+Despite the fact that the gitaly team claims strong consistency, there are possible cases when inconsistency can occur. The main goals of this technical report are to describe full gitlab  workflow of some mutator operation like "git push" and to figure out the distribute fetch process, to find possible cases when inconsistency can be hit. This research work can help us to find the most  relevant protocol to achieve strong consistency in Degitx distributed git repository manager.
```
Comment (g4s8) [inline reports/gitlab-commit-protocol/gitlab-commit-protocol.tex:81]: Please try to split long lines to multiple linse in `tex` sources, now it's not possible to comment on specific line.

> there are possible cases when inconsistency can occur

Did we really find any "inconsistency"? 

> "git push"

it should be:
```tex
``git push''
```
in sources: double ` - for opening quote, double ' - for closing.

**Review by g4s8 — CHANGES_REQUESTED** [review]
@ivanbukhtiyarov please check my comments. Also, I don't see the conclusion in this report: did we find any inconsistency and what are the risks of using this protocol in production

**Review by IlyaMoskva — APPROVED** [review]
This version is good for me. Thanks, @ivanbukhtiyarov .

**Review by g4s8 — APPROVED** [review]
@ivanbukhtiyarov thanks

**Pull request merged**